runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/endpoints/Endpoint.kt

```diff
@@ -27,9 +23,52 @@ import aws.smithy.kotlin.runtime.http.Url
  * is expected to be mutable and the client cannot modify the endpoint correctly, the operation
  * will likely fail.
  */
-public data class Endpoint(
-    public val uri: Url,
-    public val isHostnameImmutable: Boolean = false,
-) {
+public class Endpoint {
+    /**
+     * The base URL endpoint clients will use to make API calls to e.g. "api.myservice.com".
+     * NOTE: Only `scheme`, `port`, `host` `path`, and `parameters` are valid. Other URL elements are ignored.
+     */
+    public val uri: Url
+
+    /**
+     * Flag indicating whether the hostname can be modified by the SDK client.
+     */
+    public val isHostnameImmutable: Boolean
```

Comment:
Could/should this just be an attribute? (We can always have it as an extension)

Reply:
This wasn't a new field, it just looked that way because I changed the Endpoint to non-data class, that's since been reverted.

Reply:
Right yeah no what I was suggesting/wondering is if we have this new `attributes` field (especially if it's our typed property bag) we could just make this an attribute rather than an explicit field. 

e.g.



```kotlin
data class Endpoint(val url: Url, val attributes: Attributes = ...)

val Endpoint.isHostNameImmutable: Boolean
    get() = attributes[EndpointImmutable] ?: false

```

It's fine as is too I was just wondering how much we can shove into the property bag since that's effectively what this is, an attribute of the endpoint.